pachterlab/sleuth
# Improved sanity checks and testing

This first of all fixes the typo identified in #181.

Next, to build off of a comment from that thread, this pull request has added several additional tests. Work on theses tests revealed a need to improve the sanity checks already present.

Sanity checks improved / added:
+ `sleuth_fit` will stop with error if there are other fits that are not sync'ed with the transform functions (this is to force the user to rerun `sleuth_prep` and `sleuth_fit` if they want to use a new transform function)
+ If the sleuth object has some fits with counts and some fits with TPMs, changing the transform function for counts or tpms only affects the fits related to the respective variable
+ `gene_column` cannot be set if there is no `target_mapping`. It also cannot be set to a column name that doesn't exist in `target_mapping`
+ `gene_mode` and `pval_aggregate` cannot be set to anything other than `TRUE` or `FALSE`
+ `gene_mode` and `pval_aggregate` cannot be set if there is no `gene_column` set. This works correctly now.

Tests added:
+ Run with warning if there are duplicated IDs in transcript mode
+ Stop with error if there are duplicated IDs in gene mode
+ Proper sanity checks for when the user changes the transform functions (counts or TPMs) -- this is done for both the methods and the `$<-` methods
+ Proper sanity checks for when the user changes the normalization functions
+ Proper sanity checks for when the user changes `gene_column`
+ Proper sanity checks for when the user changes `gene_mode` or `pval_aggregate`
+ Check to make sure p-value aggregation works correctly.

Finally, the `README` for the test data folder ('tests/testthat/ellahi') is now updated to reflect how the test data was generated for comparisons. The test data in the folder have been analyzed using sleuth v0.30.0
